internal/services/contracts_events_consumer.go

```diff
@@ -521,3 +527,61 @@ func (c *ContractsEventsConsumer) handleSyntheticDeviceNodeBurnedEvent(ctx conte
 	_, err := sd.Delete(ctx, c.dbs.DBS().Writer)
 	return err
 }
+
+func (c *ContractsEventsConsumer) handleTokensTransferredForDevice(ctx context.Context, e *ContractEventData) error {
+	var args TokensTransferredForDeviceData
+	if err := json.Unmarshal(e.Arguments, &args); err != nil {
+		return err
+	}
+
+	reward := models.Reward{
+		IssuanceWeek:      1,
+		VehicleID:         int(args.VehicleID.Int64()),
+		ReceivedByAddress: null.BytesFrom(args.User.Bytes()),
+		EarnedAt:          e.Block.Time,
+	}
+
+	cols := models.RewardColumns
+
+	if common.HexToAddress(c.settings.AftermarketDeviceAddr) == args.DeviceNftProxy {
+		reward.AftermarketTokenID = null.IntFrom(int(args.DeviceNode.Int64()))
+		reward.AftermarketEarnings = null.IntFrom(int(args.Amount.Int64()))
+		return reward.Upsert(ctx, c.dbs.DBS().Writer, true,
+			[]string{cols.IssuanceWeek, cols.VehicleID},
+			boil.Whitelist(cols.AftermarketEarnings, cols.AftermarketTokenID),
+			boil.Infer())
+	} else if common.HexToAddress(c.settings.SyntheticDeviceAddr) == args.DeviceNftProxy {
+		reward.SyntheticTokenID = null.IntFrom(int(args.DeviceNode.Int64()))
+		reward.SyntheticEarnings = null.IntFrom(int(args.Amount.Int64()))
+		return reward.Upsert(ctx, c.dbs.DBS().Writer, true,
+			[]string{cols.IssuanceWeek, cols.VehicleID},
+			boil.Whitelist(cols.SyntheticEarnings, cols.SyntheticTokenID),
+			boil.Infer())
+	}
+
+	return nil
+}
+
+func (c *ContractsEventsConsumer) handleTokensTransferredForConnectionStreak(ctx context.Context, e *ContractEventData) error {
+	var args TokensTransferredForConnectionStreakData
+	if err := json.Unmarshal(e.Arguments, &args); err != nil {
+		return err
+	}
+
+	reward, err := models.Rewards(
```

Comment:
Must we query? I think we could construct a struct with issuance week, vehicle id, streak earning, connection streak, and call update on that.